src/sql/base/browser/ui/table/tableDataView.ts

```diff
@@ -155,43 +151,46 @@ export class TableDataView<T extends Slick.SlickData> implements IDisposableData
 		this._onRowCountChange.fire(this.getLength());
 	}
 
-	find(exp: string, maxMatches: number = 0): Thenable<IFindPosition> {
+	find(exp: string, maxMatches?: number): Promise<IFindPosition> {
 		if (!this._findFn) {
 			return Promise.reject(new Error('no find function provided'));
 		}
 		this._findArray = new Array<IFindPosition>();
 		this._findIndex = 0;
 		this._onFindCountChange.fire(this._findArray.length);
 		if (exp) {
-			this._findObs = Observable.create((observer: Observer<IFindPosition>) => {
-				for (let i = 0; i < this._data.length; i++) {
-					let item = this._data[i];
-					let result = this._findFn!(item, exp);
-					if (result) {
-						result.forEach(pos => {
-							let index = { col: pos, row: i };
-							this._findArray.push(index);
-							observer.next(index);
-							this._onFindCountChange.fire(this._findArray.length);
-						});
-						if (maxMatches > 0 && this._findArray.length > maxMatches) {
-							break;
-						}
-					}
-				}
-			});
-			return this._findObs!.take(1).toPromise().then(() => {
-				return this._findArray[this._findIndex];
+			this._startSearch(exp, maxMatches);
+			return new Promise<IFindPosition>((resolve) => {
+				const disp = this.onFindCountChange(e => {
+					resolve(this._findArray[e]);
+					disp.dispose();
```

Comment:
So if I'm reading this correctly - this is only going to ever return a single item? Do you have any insight into why we even have maxMatches then?

Reply:
Ah ok so you can call findNext/findPrevious to move around the found items after the initial find call. Ignore this comment then.